core/primitives/src/receipt.rs

```diff
@@ -90,6 +91,73 @@ pub enum Receipt {
     V1(ReceiptV1),
 }
 
+/// A receipt that is stored in the state with added metadata. A receipt may be
+/// stored in the state as a delayed receipt, buffered receipt or a promise
+/// yield receipt. The metadata contains additional information about receipt
+///
+/// Please note that the StateStoredReceipt implements custom serialization and
+/// deserialization. Please see the comment on [ReceiptOrStateStoredReceipt]
+/// for more details.
+///
+/// This struct is versioned so that it can be enhanced in the future.
+#[derive(PartialEq, Eq, Debug)]
+pub enum StateStoredReceipt {
```

Comment:
Should the new structs have the `ProtocolSchema` derive that @Longarithm  added recently? They are stored in the trie and can't be changed without a protocol change.